File: clients/batch/python/batch.py
#!/usr/bin/env python
import barrister

trans  = barrister.HttpTransport("http://localhost:3000/v1/todos")
client = barrister.Client(trans)

batch = client.start_batch()
batch.TodoManager.createTodo({ "title" : "Call Mom", "completed" : False })
batch.TodoManager.createTodo({ "title" : "Call Dad", "completed" : False })
batch.TodoManager.createTodo({ "title" : "Wash car", "completed" : False })
batch.TodoManager.createTodo({ "title" : "Eat Ham", "completed" : False })

results = batch.send()
for res in results:
    # either res.error or res.result will be set
    if res.error:
        # res.error is a barrister.RpcException, so you can raise it if desired
        print "err.code=%d" % res.error.code
    else:
        # result from a successful call
        print res.result

File: clients/batch/node/batch.js
#!/usr/bin/env node

var barrister = require('barrister');

var client = barrister.httpClient("http://localhost:3000/v1/todos");

client.loadContract(function(err) {
  var batch = client.startBatch();
  var batchTodoManager = batch.proxy("TodoManager");

  batchTodoManager.createTodo({ 'title' : 'Call Mom', 'completed' : false })
  batchTodoManager.createTodo({ 'title' : 'Call Dad', 'completed' : false })
  batchTodoManager.createTodo({ 'title' : 'Wash car', 'completed' : false })
  batchTodoManager.createTodo({ 'title' : 'Eat Ham', 'completed' : false })

  batch.send(function(err, result) {
    for (var i = 0, len = result.length; i < len; i++) {
      // either result[i].error or result[i].result will be set
      if (result[i].error) {
        console.log("err.code=" + result[i].error.code);
      }
      else {
        // result from a successful call
        console.log(result[i].result);
      }
    }
  });
});
